### PR TITLE
Move QIR driver generator

### PR DESCRIPTION
**Pull request by cesarzc**
This change moves the implementation of the QIR driver generator from the qsharp-compiler repo.

Additionally, it extends the functionality to make it easier to create drivers for other simulators.

### CODE REVIEW

File: src/Qir/Execution/Tools/Driver/QirFullStateSimulatorInitializer.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft Corporation.
```
Comment (achocron) [inline src/Qir/Execution/Tools/Driver/QirFullStateSimulatorInitializer.cs:1]: What is this initializer concept? What was this moved from?

File: src/Qir/Execution/Tools/Microsoft.Quantum.Qir.Tools.csproj [hunk 2]
```diff
@@ -31,7 +31,7 @@
 
   <ItemGroup>
     <Compile Update="Driver\QirCppDriver.cs">
-      <DesignTime>False</DesignTime>
```
Comment (achocron) [inline src/Qir/Execution/Tools/Microsoft.Quantum.Qir.Tools.csproj]: Was this change intentional?

File: src/Qir/Execution/Tests.Microsoft.Quantum.Qir.Tools/QirDriverGeneratorTests.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+                    new EntryPointOperation
+                    {
+                        Name = "UseResultArg",
+                        Parameters = new List<Parameter>{new Parameter{ Name = "ResultArg", Type = DataType.ResultType}}
+                    }
+                }
+            };
+
+        [Theory]
+        [InlineData("UseNoArgs")]
```
Comment (achocron) [inline src/Qir/Execution/Tests.Microsoft.Quantum.Qir.Tools/QirDriverGeneratorTests.cs:37]: Why not use all of the test cases?

**Review by achocron — APPROVED** [review]
Looks good, I think it's good to merge once the rest of the test cases are added.

File: src/Qir/Execution/Tests.Microsoft.Quantum.Qir.Tools/QirDriverGeneratorTests.cs (hunk 3 above)
Comment (cesarzc) [reply src/Qir/Execution/Tests.Microsoft.Quantum.Qir.Tools/QirDriverGeneratorTests.cs:146]: Added all the test cases.

File: src/Qir/Execution/Tools/Driver/QirFullStateSimulatorInitializer.cs (hunk 1 above)
Comment (cesarzc) [reply src/Qir/Execution/Tools/Driver/QirFullStateSimulatorInitializer.cs:1]: Chatted offline. In summary, this was not moved, it's a new concept to enable creating drivers for multiple simulators and not only the full state one.
